src/main/java/net/sf/jabref/logic/util/version/VersionInformation.java

```diff
@@ -0,0 +1,184 @@
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+package net.sf.jabref.logic.util.version;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.time.LocalDateTime;
+
+import javax.swing.JOptionPane;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefGUI;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.gui.help.NewVersionDialog;
+import net.sf.jabref.logic.l10n.Localization;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONObject;
+
+public class VersionInformation {
+
+    private static final Log LOGGER = LogFactory.getLog(VersionInformation.class);
+
+    private final static String JABREF_GITHUB_URL = "https://api.github.com/repos/JabRef/jabref/releases/latest";
+    public final static String JABREF_DOWNLOAD_URL = "http://www.fosshub.com/JabRef.html";
+
+    private Version latestVersion;
+
+
+    /**
+     * this method is meant for a background versions check as it respects if the user previously ignored an update or if it has already been checked within this JabRef instance
+     * The check is started within a Thread and will show a popup after some time if a new Version is available and update the preferecesDialog
+     */
+    public void checkForUpdatesAutomatically() {
+        checkForUpdates(false, true, false);
+    }
+
+    /**
+     * This method should be used when the user explicitly searches for an update as it ignores if the user previuosly ignored an update or if it has already been checked in this JabRef instance
+     * The check is started within a Thread and will show a popup after some time if a new Version is available and update the preferecesDialog
+     */
+    public void checkForUpdatesManually() {
+        checkForUpdates(true, false, true);
+    }
+
+    private void checkForUpdates(boolean forceCheck, boolean respectIgnore, boolean manualExecution) {
+        new Thread() {
+            @Override
+            public void run() {
+                boolean newerVersion = isNewerVersionAvailable(forceCheck, respectIgnore, manualExecution);
+                if (newerVersion) {
+                    new NewVersionDialog();
+                }
+                else if (manualExecution){
+                    JOptionPane.showMessageDialog(JabRefGUI.getMainFrame(),
+                            Localization.lang("uptodate"),
+                            Localization.lang("uptodate"),
+                            JOptionPane.INFORMATION_MESSAGE);
+                }
+                JabRefGUI.getMainFrame().updatePreferenceDialog();
+            }
+        }.start();
```

Comment:
Threads should be started via the JabRefExecutorService. But if you want to update the UI upon completion, you need a SwingWorker which ensures that one part is executed outside the EDT, and the second part is executed on the EDT. Another issue here is that this class in the logic package, but makes use of GUI classes - this is forbidden.